midealocal/decrypt.py

```
@@ -0,0 +1,53 @@
+from Crypto.Cipher import AES
+import binascii
+import hashlib
+import sys
+
+def decrypt_file(input_file, key):
+    # Generate MD5 hash and take first 16 chars
+    key_bytes = hashlib.md5(key.encode()).hexdigest()[:16].encode()
```

Ruff (0.16.0)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)